picasso/src/main/java/com/squareup/picasso3/Picasso.java

```diff
@@ -231,6 +254,26 @@ public void cancelTag(@NonNull Object tag) {
     }
   }
 
+  @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
```

Comment:
have never used this library. super curious. can't figure out why `LifecycleObserver` is a marker interface that tells `Lifecycling` to look up annotated methods reflectively instead of having the interface implement the methods. seems like the worst of both options.

Reply:
Tbh, didn't dig into this too much, went by the recommendation here: https://developer.android.com/reference/android/arch/lifecycle/Lifecycle

Looking at the source now though: DefaultLifecycleObserver has default methods (so Java 8 only), but a dummy impl of FullLifecycleObserver would also work on Java 7.